src/query/plan/operator.cpp

```diff
@@ -405,11 +406,28 @@ class ScanAllCursor : public Cursor {
       vertices_it_.emplace(vertices_.value().begin());
     }
 
+#ifdef MG_ENTERPRISE
+    FilterNodes(context.fine_grained_access_checker, context.db_accessor);
+    if (vertices_it_.value() == vertices_.value().end()) return false;
+#endif
+
     frame[output_symbol_] = *vertices_it_.value();
     ++vertices_it_.value();
     return true;
   }
 
+  void FilterNodes(const FineGrainedAccessChecker *fine_grained_access_checker, DbAccessor *db_accessor) {
+    if (!fine_grained_access_checker) return;
```

Comment:
is it better to keep this if outside the method?

Reply:
and then just leave the logic in the method, that actually does the filtering

Reply:
Maybe it would be good if this method was returning a vertex. that way we could even test it (if that's even possible). 
if iterator is exhausted, return null, else return a vertex.
The method would be then called GetVertex or something like that?
And yes, we can maybe consider calling here everything as Vertex instead of nodes since here it is mostly about vertices naming in the operator

Reply:
it's just my preference that I don't like methods that return void :') you do as you please